lib/Dialect/RTL/Ops.cpp

```diff
@@ -806,19 +832,25 @@ static LogicalResult verifyUTVariadicRTLOp(Operation *op) {
 }
 
 OpFoldResult AndOp::fold(ArrayRef<Attribute> operands) {
-  auto size = inputs().size();
+  auto width = getType().cast<IntegerType>().getWidth();
+  APInt value(/*numBits=*/width, -1, /*isSigned=*/false);
 
-  // and(x) -> x -- noop
-  if (size == 1u)
-    return inputs()[0];
-
-  APInt value;
+  // and(x, 0, 1) -> 0 -- annulment
+  for (auto operand : operands) {
+    if (!operand)
+      continue;
+    value &= operand.cast<IntegerAttr>().getValue();
+    if (value.isNullValue())
+      return getIntAttr(value, getContext());
+  }
 
-  // and(..., 0) -> 0 -- annulment
-  if (matchPattern(inputs().back(), m_RConstant(value)) && value.isNullValue())
-    return inputs().back();
+  // and(x, x, x) -> x -- noop
+  if (llvm::all_of(inputs(), [&](auto in) { return in == inputs()[0]; }))
+    return inputs()[0];
 
-  return {};
+  // Constant fold
+  return constFoldVariadicOp<IntegerAttr>(
+      operands, [](APInt a, APInt b) { return a & b; });
```

Comment:
nit but APInt types aren't trivial, I'd recommend `[](const APInt &a, const APInt &b)`

Reply:
Any preference between these two?
```c++
[](const APInt &a, const APInt &b) { return a & b; }
```
```c++
[](APInt &a, const APInt &b) { a &= b; }
```

Reply:
I guess I'd go with the second one, but at this point it's quite a micro optimization.  Both approaches work for me!   Feel free to land in whichever config you'd like.  Thanks @youngar !